From 7af443a1c6a041dec6fb0f1c962422aa9e42fb2c Mon Sep 17 00:00:00 2001
From: SignKirigami <prcups@krgm.moe>
Date: Tue, 25 Aug 2026 11:01:14 +0800
Subject: [PATCH 5/6] AOSCOS: touchpad: forward X11 save state changes

Forward the device's needsSaveChanged signal through XlibBackend so the
KCM enables Apply after settings change, including for hotplugged devices.

Use gpt-5.6-sol to generate fix code.

Signed-off-by: SignKirigami <prcups@krgm.moe>
---
 kcms/touchpad/backends/x11/xlibbackend.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kcms/touchpad/backends/x11/xlibbackend.cpp b/kcms/touchpad/backends/x11/xlibbackend.cpp
index 74a3919da4..83b70deffa 100644
--- a/kcms/touchpad/backends/x11/xlibbackend.cpp
+++ b/kcms/touchpad/backends/x11/xlibbackend.cpp
@@ -78,6 +78,8 @@ XlibBackend::XlibBackend(QObject *parent)
     m_device.reset(findTouchpad());
     if (!m_device) {
         m_errorString = i18n("No touchpad found");
+    } else {
+        connect(m_device.get(), &LibinputTouchpad::needsSaveChanged, this, &TouchpadBackend::needsSaveChanged);
     }
 }
 
@@ -200,6 +202,7 @@ void XlibBackend::devicePlugged(int /*device*/)
     if (!m_device) {
         m_device.reset(findTouchpad());
         if (m_device) {
+            connect(m_device.get(), &LibinputTouchpad::needsSaveChanged, this, &TouchpadBackend::needsSaveChanged);
             qWarning() << "Touchpad reset";
             // We get called by m_notifications, need to use deleteLater
             m_notifications.release()->deleteLater();
-- 
2.55.0

